static/app/components/smartSearchBar/index.tsx

```diff
@@ -214,6 +214,11 @@ type Props = WithRouterProps &
      * as the stream view where it is a top level concept
      */
     excludedTags?: string[];
+    /**
+     * A function that returns a warning message for a given filter key
+     * will only show a render a warning if the value is truthy
+     */
+    getFilterWarning?: (key) => React.ReactNode | null;
```

Comment:
```suggestion
    getFilterWarning?: (key) => React.ReactNode;
```

`ReactNode` already includes `null`